File: backend/src/models/Team.js
import mongoose from 'mongoose';

const teamSchema = new mongoose.Schema({
  code:     { type: String, required: true, unique: true },
  name:     { type: String, required: true },
  roomId:   { type: mongoose.Schema.Types.ObjectId, ref: 'Room', required: true },
  leaderId: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true },
  members:  [{
    userId:   { type: mongoose.Schema.Types.ObjectId, ref: 'User' },
    username: String,
    isReady:  { type: Boolean, default: false },
  }],
  score:   { type: Number, default: 0 },
  maxSize: { type: Number, default: 4 },
  isReady: { type: Boolean, default: false },
}, { timestamps: true });

export default mongoose.model('Team', teamSchema);

File: backend/dsa-multiplayer/src/main/java/com/clashcode/dsa_multiplayer/team/entity/Team.java
package com.clashcode.dsa_multiplayer.team.entity;

import com.clashcode.dsa_multiplayer.auth.entity.User;
import com.clashcode.dsa_multiplayer.room.entity.Room;
import jakarta.persistence.*;
import lombok.*;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.time.Instant;
import java.util.UUID;

@Entity
@Table(name = "teams")
@EntityListeners(AuditingEntityListener.class)
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Team {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(nullable = false, unique = true, length = 6)
    private String code;

    @Column(nullable = false, length = 100)
    private String name;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "room_id", nullable = false)
    private Room room;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "leader_id", nullable = false)
    private User leader;

    @Builder.Default
    @Column(nullable = false)
    private int score = 0;

    @Builder.Default
    @Column(name = "max_size", nullable = false)
    private int maxSize = 4;

    @Builder.Default
    @Column(name = "is_ready", nullable = false)
    private boolean isReady = false;

    @CreatedDate
    @Column(name = "created_at", nullable = false, updatable = false)
    private Instant createdAt;

    @LastModifiedDate
    @Column(name = "updated_at", nullable = false)
    private Instant updatedAt;
}
